src/codewhisperer/tracker/codewhispererCodeCoverageTracker.ts

```diff
@@ -188,10 +210,18 @@ export class CodeWhispererCodeCoverageTracker {
         }
     }
 
+    public activateTrackerIfNotActive(globals: vscode.Memento) {
+        const isTermsAccepted = globals.get<boolean>(CodeWhispererConstants.termsAcceptedKey) || false
+        if (this._isTrackerActive) return
+        if (!TelemetryHelper.instance.isTelemetryEnabled() || !isTermsAccepted) return
+        this._isTrackerActive = true
+    }
```

Comment:
I think this refactor had the right direction but could use some refinement in terms of usage. I'd put the `Memento` parameter back into the constructor and replace `_isTrackerActive` with this:
```suggestion
    public get isActive() {
        const isTermsAccepted = this.globals.get<boolean>(CodeWhispererConstants.termsAcceptedKey) || false
        
        return TelemetryHelper.instance.isTelemetryEnabled() && isTermsAccepted
    }
```

Then for ergonomics make `getTracker` use a default:
```ts
    public static getTracker(language: string, memento = globals.context.globalState): CodeWhispererCodeCoverageTracker | undefined {
```

The reasoning is that it's too easy to forget to call `activateTrackerIfNotActive` after calling `getTracker`.

Reply:
noted, yea let me update this